src/ripple/ledger/impl/View.cpp

```diff
@@ -34,6 +34,126 @@
 
 namespace ripple {
 
+namespace detail {
+
+template <
+    class V,
+    class N,
+    class = std::enable_if_t<
+        std::is_same_v<std::remove_cv_t<N>, SLE> &&
+        std::is_base_of_v<ReadView, V>>>
```

Comment:
Consider removing the `enable_if`. Rational: we aren't using `enable_if` to resolve an overload, we're using it as a type-check and as documentation. I'd probably leave it off entirely and add a comment. This is a `detail` function only called from this file (it's not on a public interface where clients can call it with weird parameter types and wonder what's wrong). If the wrong types are passed in this file, we'll still get a compile error and quickly figure out why. 

If you don't like the comment, as an alternative, consider adding a `static_assert`. I'd prefer we didn't increase our use of `enable_if` to cases like this. When we see `enable_if` I want to think "this is part of an overload set".